### ISSUE TITLE
false negative with underscore on parameter name

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#9447]
The following code reports no errors with mypy but will fail at runtime:
```py
class Base:
    def a(self, a: int) -> None: ...

class Derived(Base):
    def a(self, _a: int) -> None:  ...

def foo(b: Base) -> None:
    b.a(a=1)

foo(Derived())
```

**Comment by KotlinIsland** [comment]
Same thing with double underscore on parameter name

**Comment by gvanrossum** [comment]
This is a tough one. The problem is that this kind of error is pretty common in class hierarchies, and if we were to report `Derived.a` as an error because the parameter names don't match, I'm sure we'd get quite a few complaints. So I'm not sure what to do except propose that maybe this ought to be yet another strictness flag (but we already have too many of those).

PS. There's nothing special about underscores here -- mypy just doesn't care about the argument names. For inheritance it just checks whether the type of `Base.a` reduced to what can be expressed using `Callable` is a supertype of the type of `Derived.a` when reduced to a `Callable`.
